components/raftstore/src/store/fsm/store.rs

```diff
@@ -1495,32 +1533,20 @@ impl<'a, T: Transport, C: PdClient> StoreFsmDelegate<'a, T, C> {
     /// return false to indicate that target peer is in invalid state or
     /// doesn't exist and can't be created.
     fn maybe_create_peer(&mut self, region_id: u64, msg: &RaftMessage) -> Result<bool> {
-        let target = msg.get_to_peer();
-        if !is_initial_msg(msg.get_message()) {
-            let msg_type = msg.get_message().get_msg_type();
-            debug!(
-                "target peer doesn't exist, stale message";
-                "target_peer" => ?target,
-                "region_id" => region_id,
-                "msg_type" => ?msg_type,
-            );
-            self.ctx.raft_metrics.message_dropped.stale_msg += 1;
+        if self
+            .ctx
+            .engines
+            .kv
+            .get_value_cf(CF_RAFT, &keys::region_state_key(region_id))?
+            .is_some()
```

Comment:
It is possible that there are existing tombstone `RegionLocalState` (not written by split).